src/librc/librc-misc.c

```diff
@@ -447,3 +450,44 @@ rc_conf_value(const char *setting)
 
 	return rc_config_value(rc_conf, setting);
 }
+
+bool rc_export_variable(const char *service, const char *name, const char *value) {
+	int dirfd = openat(rc_dirfd(RC_DIR_ENVIRONMENT), service, O_RDONLY | O_DIRECTORY);
+	FILE *envfile;
+
+	if (dirfd == -1)
+		return false;
+
+	if (flock(dirfd, LOCK_EX) == -1) {
+		close(dirfd);
+		return false;
+	}
+
+	if (!value) {
+		unlinkat(dirfd, name, 0);
+		return true;
+	}
+
+	envfile = do_fopenat(dirfd, name, O_WRONLY | O_TRUNC | O_CREAT);
+	fputs(value, envfile);
+	return fclose(envfile) == 0;
```

Comment:
You also need to check the `fputs` return value. Also, shouldn't we check of `envfile` being `NULL` as well?